test/testutil/driver.c

```diff
@@ -436,13 +436,33 @@ char *test_mk_file_path(const char *dir, const char *file)
     const char *sep = "/";
 # else
     const char *sep = "";
+    char *dir_end;
+    char dir_end_sep;
 # endif
     size_t len = strlen(dir) + strlen(sep) + strlen(file) + 1;
     char *full_file = OPENSSL_zalloc(len);
 
     if (full_file != NULL) {
         OPENSSL_strlcpy(full_file, dir, len);
+# ifdef OPENSSL_SYS_VMS
+        /*
+         * If |file| contains a directory spec, we need to do some careful
+         * merging.
+         * "vol:[dir.dir]" + "[.certs]sm2-root.crt" should become
+         * "vol:[dir.dir.certs]sm2-root.crt"
+         */
+        dir_end_sep = *( dir_end = &full_file[strlen(full_file) - 1] );
```

Comment:
Break this monstrosity in half please, *& is a bit of a silly combination:
```c
dir_end = &full_file[strlen(full_file) - 1];
dir_end_sep = *dir_end;
```
The alternative `dir_end_sep = *(dir_end = full_file + strlen(full_file) - 1);` is okay bit still a bit less readable IMO.

Reply:
I consider the alternative even more monstrous :wink:

Reply:
Split done

Reply:
Consensus reached :grin:

Reply:
> Split done

You've probably forgot to push @levitte

Reply:
I eventually did